ttnn/cpp/ttnn/operations/ccl/all_gather/device/multi_core/all_gather_op_multi_core.cpp

```diff
@@ -604,18 +780,38 @@ operation::ProgramWithCallbacks all_gather_multi_core_with_workers_helper(
                 log_trace(tt::LogOp,"\tlast_output_page_offset={}", last_output_page_offset);
                 log_trace(tt::LogOp,"\tlast_output_addr_offset={}", last_output_addr_offset);
 
-                if (!is_linear || !is_last_chip_in_chain) {
+                bool sender_enabled = (!is_linear || !is_last_chip_in_chain);
+                bool receiver_enabled = (!is_linear || !is_first_chip_in_chain);
+
+                // SEND WORKERS
+                if (!sender_enabled) {
```

Comment:
Since these are known - we should ideally just feed these into the functions that generate the core range sets. IMO we shouldn't have this pattern where we invoke a kernel intentionally knowing it's a NOP. It's a lot of extra code and branches and some extra args, and then it's also harder to understand why this is the way it is.